logbook-core/src/main/java/org/zalando/logbook/ChunkingHttpLogWriter.java

```diff
@@ -0,0 +1,51 @@
+package org.zalando.logbook;
+
+import org.zalando.logbook.DefaultLogbook.SimpleCorrelation;
+import org.zalando.logbook.DefaultLogbook.SimplePrecorrelation;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.Math.min;
+
+public final class ChunkingHttpLogWriter implements HttpLogWriter {
+
+    private final int size;
+    private final HttpLogWriter writer;
+
+    public ChunkingHttpLogWriter(final int size, final HttpLogWriter writer) {
+        this.size = size;
+        this.writer = writer;
+    }
+
+    @Override
+    public boolean isActive(final RawHttpRequest request) throws IOException {
+        return writer.isActive(request);
+    }
+
+    @Override
+    public void writeRequest(final Precorrelation<String> precorrelation) throws IOException {
+        for (final String part : split(precorrelation.getRequest())) {
+            writer.writeRequest(new SimplePrecorrelation<>(precorrelation.getId(), part));
+        }
+    }
+
+    @Override
+    public void writeResponse(final Correlation<String, String> correlation) throws IOException {
+        for (final String part : split(correlation.getResponse())) {
+            writer.writeResponse(new SimpleCorrelation<>(correlation.getId(), correlation.getRequest(), part));
+        }
+    }
+
+    private Iterable<String> split(final String s) {
```

Comment:
Still creating the whole list up front, I was thinking about something like

    private void splitted(final String string, Consumer<String> consumer) throws IOException {
        final char[] buf = new char[size];
        int len;
        try (final StringReader r = new StringReader(string)) {
            while (-1 != (len = r.read(buf))) {
                consumer.accept(new String(buf, 0, len));
            }
        }
    }

Although you would need a `Consumer` interface that is allowed to throw `IOException` to make it this generic. Or just inline the code twice.